lib/engine/ability/exchange.rb

```diff
@@ -7,9 +7,12 @@ module Ability
     class Exchange < Base
       attr_reader :corporation, :from
 
-      def setup(corporation:, from:)
+      def setup(from:, corporation: nil, corporations: nil)
         @corporation = corporation
+        @corporations = corporations
```

Comment:
@corporation = corporation || corporations[0]
@corporations = corporations || [@corporation]

Reply:
That would have been better. Have now changed to just using corporations which removed this need.